autorest/azure/azure.go

```diff
@@ -28,6 +28,19 @@ const (
 	HeaderRequestID = "x-ms-request-id"
 )
 
+// AADServiceError encapsulates the error response from Azure Token Service.
+type AADServiceError struct {
+	ErrorTitle       string `json:"error"`
+	ErrorDescription string `json:"error_description"`
+	ErrorCodes       []int  `json:"error_codes"`
+	TraceID          string `json:"trace_id"`
+	CorrelationID    string `json:"correlation_id"`
+}
+
+func (aade *AADServiceError) Error() string {
```

Comment:
it seems without pointers corresponds to const functions. I will make that change